### PR TITLE
Raising registry exception for adding autoscaler policy and dep policy

### PR DESCRIPTION
**Pull request by Vishanth**
Raising registry exception  for persisting autoscaler policy and persisting deployment policy.

### CODE REVIEW

File: components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/registry/RegistryManager.java [hunk 1]
```diff
@@ -118,19 +118,33 @@ private void persist(Object dataObj, String resourcePath) throws AutoScalerExcep
...
+                    AutoscalerConstants.AS_POLICY_RESOURCE + "/" + autoscalePolicy.getId();
+
+            persist(autoscalePolicy, resourcePath);
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Autoscaler policy written to registry: [id] %s [name] %s [description] %s",
+                        autoscalePolicy.getId(), autoscalePolicy.getDisplayName(), autoscalePolicy.getDescription()));
+            }
+        } catch (Exception e) {
+            log.error((String.format("Unable to persist autoscaler policy [autoscaler-policy-id] %s",
+                    autoscalePolicy.getId())), e);
```
Comment (imesh) [inline components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/registry/RegistryManager.java]: Thanks for looking into this @Vishanth! IMO we need to raise an Autoscaler exception if there is a problem in persisting an artifact. Would you mind throwing an Autoscaler exception something like AutoscalerPersistenceException from this method?
Thanks

Comment (imesh) [inline components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/registry/RegistryManager.java]: @Vishanth This pull request was open for quite some time now. Shall we proceed further on this? If you could try to re-create the original issue and verify whether this change fix that problem it would be great!
Thanks

Comment (Vishanth) [inline components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/registry/RegistryManager.java]: @imesh There was a problem with my Git. I will make sure this solves the problem and send a fresh PR. Shall we close this now?

Comment (imesh) [inline components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/registry/RegistryManager.java]: Yes please close this as we have done the same fix in a different pull request.
